content/2015.05.cliquet-permissions-en.rst

```diff
@@ -0,0 +1,535 @@
+| ``record:<page id>``    | ``read``              | ``Everyone``                |
++-------------------------+-----------------------+-----------------------------+
+
+
+The HTTP API
+------------
+
+During the creation of an object, the user is given the ``write`` permission on
+the object:
+
+.. code-block:: http
+
+    PUT /v1/buckets/servicedenuages_blog HTTP/1.1
+    Authorization: Bearer 0b9c2625dc21ef05f6ad4ddf47c5f203837aa32ca42fced54c2625dc21efac32
+    Accept: application/json
+
+    HTTP/1.1 201 Created
+    Content-Type: application/json; charset=utf-8
+
+    {
+        "id": "servicedenuages_blog",
+        "permissions": {
+            "write": ["fxa:49d02d55ad10973b7b9d0dc9eba7fdf0"]
+        }
+    }
+
+It's possible to add permissions using the ``PATCH`` HTTP method.
+
+.. code-block:: http
+
+    PATCH /v1/buckets/servicedenuages_blog/collections/articles HTTP/1.1
+    Authorization: Bearer 0b9c2625dc21ef05f6ad4ddf47c5f203837aa32ca42fced54c2625dc21efac32
+    Accept: application/json
+
+    {
+        "permissions": {
+            "read": ["+system.Everyone"]
+        }
+    }
+
+    HTTP/1.1 201 Created
+    Content-Type: application/json; charset=utf-8
+
+    {
+        "id": "servicedenuages_blog",
+        "permissions": {
+            "write": ["fxa:49d02d55ad10973b7b9d0dc9eba7fdf0"],
+            "read": ["system.Everyone"]
+        }
+    }
+
+For ``PATCH``, we are thinking about using a syntax prefixed with a ``+`` or
+a ``-`` to add or remove *principals* on an ACL.
+
+It is also possible to do a ``PUT`` to reset the ACLs, knowing that the ``PUT``
+will then add the current user to the list. It's possible to use a ``PATCH`` to
+remove herself from the list. Adding the current user allows to avoid
+situations where nobody has access to the data anymore.
+
+.. note:: 
+
+    The ``create`` permission is used for a ``POST`` but also for a ``PUT``
+    when the record doesn't exist.
+
+
+The specific case of user data
+------------------------------
+
+One of the current feature of *Kinto* is to handle record collections by user.
+
+On *\*nix* systems, it's possible, for an application, to save configuration
+for the current user in her personal folder, without bothering about its
+specific location, using ``~``.
+
+In our case, if an application want to save contacts for an user, it can use
+a shortcut to reference the *personal* bucket of the user:
+``/buckets/personal/collections/contacts``.
+
+This URL will return a ``HTTP 307`` to the current user *bucket*.
+
+.. code-block:: http
+
+    POST /v1/buckets/personal/collections/contacts/records HTTP/1.1
+
+    {
+       "name": "Rémy",
+       "emails": ["remy@example.com"],
+       "phones": ["+330820800800"]
+    }
+
+    HTTP/1.1 307 Temporary Redirect
+    Location: /v1/buckets/fxa:49d02d55ad10973b7b9d0dc9eba7fdf0/collections/contacts/records
+
+Like so, it's possible for Alice to share her contacts with Bob. She just have
```

Comment:
So it seems you've actually named your user already :) we should do this at the beginning of the article.